coral-spark/src/main/java/com/linkedin/coral/spark/containers/SparkUDFInfo.java

```diff
@@ -94,4 +95,22 @@ public String toString() {
     return "SparkUDFInfo{" + "className='" + className + '\'' + ", functionName='" + functionName + '\''
         + ", artifactoryUrls='" + artifactoryUrls + '\'' + ", udfType='" + udfType + '\'' + '}';
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (o == null || getClass() != o.getClass()) {
+      return false;
+    }
+    SparkUDFInfo that = (SparkUDFInfo) o;
+    return Objects.equals(className, that.className) && Objects.equals(functionName, that.functionName)
+        && Objects.equals(artifactoryUrls, that.artifactoryUrls) && udfType == that.udfType;
+  }
+
+  @Override
+  public int hashCode() {
```

Comment:
you could use IDE to generate both equal and hashCode method btw. I don't think there are problems in this implementation but just FYI.

Reply:
Yeah these 2 methods were auto-generated by IntelliJ.